docs/configuration/unit_testing_rules.md

```diff
@@ -0,0 +1,213 @@
+sort_rank: 6
+---
+
+# Unit Testing for Rules
+
+You can use `promtool` to test your rules.
+
+```shell
+# For a single test file.
+./promtool test rules test.yml
+
+# If you have multiple test files, say test1.yml,test2.yml,test2.yml
+./promtool test rules test1.yml test2.yml test3.yml
+```
+
+## Test file format
+
+```yaml
+# This is a list of rule files to consider for testing.
+rule_files:
+  [ - <file_name> ]
+
+# optional, default = 1m
+evaluation_interval: <duration>
+
+# The order in which group names are listed below will be the order of evaluation of 
+# rule groups (at a given evaluation time). The order is guaranteed only for the groups mentioned below. 
+# All the groups need not be mentioned below.
+group_eval_order:
+  [ - <group_name> ]
+
+# All the tests are listed here.
+tests:
+  [ - <test_group> ]
+```
+
+### `<test_group>`
+
+``` yaml
+# Series data
+interval: <duration>
+input_series:
+  [ - <series> ]
+
+# Unit tests for the above data.
+
+# Unit tests for alerting rules. We consider the alerting rules from the input file.
+alert_rule_test:
+  [ - <alert_test_case> ]
+
+# Unit tests PromQL expressions.
+promql_expr_test:
+  [ - <promql_test_case> ]
+```
+
+### `<series>`
+
+```yaml
+# This follows the usual series notation (x{a="b", c="d"}).
+series: <string>
+
+# This uses expanding notation.
+# Expanding notation: 
+#     a+bxc ~ a a+b a+(2*b) a+(3*b) … a+(c*b)
+#     a-bxc ~ a a-b a-(2*b) a-(3*b) … a-(c*b)
+# Examples:
+#     1. -2+4x3 ~ -2 2 6 10
+#     2.  1-2x4 ~ 1 -1 -3 -5 -7
+values: <string>
+```
+
+### `<alert_test_case>`
+
+Prometheus allows you to have same alertname for different alerting rules. Hence in this unit testing, you have to list the union of all the firing alerts for the alertname under a single `<alert_test_case>`.
+
+``` yaml
+# It's the time elapsed from time=0s when the alerts have to be checked.
+eval_time: <duration>
+
+# Name of the alert to be tested.
+alertname: <string>
+
+# List of expected alerts which are firing under the given alertname at 
+# given evaluation time. If you want to test if an alerting rule should 
+# not be firing, then you can mention the above fields and leave 'exp_alerts' empty.
+exp_alerts:
+  [ - <alert> ]
+```
+
+### `<alert>`
+
+Remember, this alert shoud be firing.
+
+``` yaml
+# These are the expanded labels and annotations of the expected alert. 
+# Note: labels also include the labels of the sample associated with the 
+# alert (same as what you see in `/alerts`, without series `__name__` and `alertname`)
+exp_labels:
+  [ <labelname>: <string> ]
+exp_annotations:
+  [ <labelname>: <string> ]
+```
+
+### `<promql_test_case>`
+
+```yaml
+# Expression to evaluate
+expr: <string>
+
+# It's the time elapsed from time=0s when the alerts have to be checked.
+eval_time: <duration>
+
+# Expected samples at the given evaluation time.
+exp_samples:
+  [ - <sample> ]
+```
+
+### `<sample>`
+
+```yaml
+# Labels of the sample in series notation.
+labels: <series_notation>
```

Comment:
What's series notation?

Reply:
I updated it with examples. I meant to say `<metric name>{<label name>=<label value>, ...}`.